src/site/content/en/notifications/use-push-notifications-to-engage-users/index.md

```diff
@@ -0,0 +1,161 @@
+---
+layout: post
+title: Use push notifications to engage and re-engage users
+authors:
+  - katejeffreys
+description: |
+  Use push notifications to engage users with targeted, timely updates.
+date: 2019-10-21
+codelabs:
+  - codelab-notifications-get-started
+  - codelab-notifications-service-worker
+  - codelab-notifications-web-push
+---
+
+## Why use push notifications?
+
+Notifications present small chunks of information to a user, letting web apps provide targeted updates.
+
+Web push technology lets a server provide notifications to a client at any time–even when the client is inactive.
+
+Push notifications can help users to get the most out of your app by prompting them to re-open it and use it based on the latest information. 
+
+## Creating and sending notifications
+
+Create notifications using the [Notifications API](https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API). A `Notification` object has a `title` string and an `options` object. For example:
+
+```js
+let title = 'Hi!';
+let options = { 
+  body: 'Very Important Message',
+  /* other options can go here */
+};
+let notification = new Notification(title, options);
+```
+
+The `title` is displayed in bold when the notification is active, while the `body` contains additional text.
+
+### Get permission to send notifications
+
+To display a notification, your app must get permission from the user to do so: 
+
+```js
+Notification.requestPermission();
+```
+
+{% Aside %}
+
+The web platform encourages developers to request notifications permission when users understand what is being asked and why. 
+
+Bring up the permission prompt **in response to a user gesture** and **when the value of notifications is obvious**. Avoid bringing up the permission prompt immediately when the user lands on the page.
+
+{% endAside %}
+
+## How do push notifications work?
+
+The real power of notifications comes from the combination of service workers and push technology.
+
+*   Service workers can run in the background and display notifications even when your app itself is not visible on screen. 
+
+*   Push technology lets a server decide when to send notifications. A push service creates unique URLs for each subscribed service worker. Sending messages to the URL raises events on the service worker.
+
+In the following example flow, the client and service worker use vanilla JavaScript with no extra libraries. The server is built with the [`express` npm package](https://www.npmjs.com/package/express) on [Node.js](https://nodejs.org/en/), and uses the [`web-push` npm package](https://www.npmjs.com/package/web-push) to send notifications. To send information to the server, the client makes a call to a POST URL that the server has exposed.
```

Comment:
I think this paragraph, explaining how the example is built, is really helpful. Might also be helpful for folks new to service workers and push to include a sentence or two at the start of the para explaining what the example _does_? I know that's conveyed in the headings and implied by what's come before, but FWIW I think a little repetition would be good to make the post structure super obvious. Feel free to ignore if you disagree!